qiskit/chemistry/results/eigenstate_result.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Eigenstate results module."""
+
+from typing import Optional, List, Union
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Instruction
+from qiskit.quantum_info import Statevector
+from qiskit.result import Result
+from qiskit.aqua.algorithms import AlgorithmResult
+from qiskit.aqua.operators import OperatorBase
+
+
+class EigenstateResult(AlgorithmResult):
+    """The eigenstate result interface."""
+
+    @property
+    def eigenenergies(self) -> Optional[np.ndarray]:
+        """ returns eigen energies """
+        return self.get('eigenenergies')
+
+    @eigenenergies.setter
+    def eigenenergies(self, value: np.ndarray) -> None:
+        """ set eigen energies """
+        self.data['eigenenergies'] = value
+
+    @property
+    def eigenstates(self) -> Optional[List[Union[str, dict, Result, list, np.ndarray, Statevector,
+                                                 QuantumCircuit, Instruction, OperatorBase]]]:
+        """ returns eigen states """
+        return self.get('eigenstates')
+
+    @eigenstates.setter
+    def eigenstates(self, value: List[Union[str, dict, Result, list, np.ndarray, Statevector,
+                                            QuantumCircuit, Instruction, OperatorBase]]) -> None:
+        """ set eigen states """
+        self.data['eigenstates'] = value
+
+    @property
+    def groundenergy(self) -> Optional[float]:
+        """ returns ground energy """
+        energies = self.get('eigenenergies')
+        if energies:
+            return energies[0].real
```

Comment:
This condition was true for the excited state solvers that were/are here I think. In general it seems that this assumption could fail if the excited states algo does not include ground state. Just noting for future consideration rather than any change at present.

Reply:
Excited states calculation should always include a ground state calculation. It is either bases on the Ground State Calculation (in EOM case) or the ground state is just one state of the states calculated by an exact eigensolver
 (numpy)

Reply:
You are citing what we have currently and yes today its true.